### PR TITLE
Windows handles metric

### PR DESCRIPTION
**Pull request by avilevy18**
## Description

Add windows handle metric to host metric receiver 
## Related issue

[b/279901505](http://b/279901505)
## How has this been tested?

Integration and unit tests
## Checklist:
- Unit tests
  - [ ] Unit tests do not apply.
  - [x] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [ ] This PR introduces no user visible changes.
  - [x] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [ ] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: apps/hostmetrics.go [hunk 1]
```diff
@@ -301,6 +311,11 @@ func (r MetricsReceiverHostmetrics) Pipelines(_ context.Context) []otel.Receiver
 					otel.ToggleScalarDataType,
 					otel.AddLabel("process", "all"),
 				),
+				otel.RenameMetric(
```
Comment (igorpeshansky) [inline apps/hostmetrics.go:314]: Should this also happen only on Windows?

Comment (avilevy18) [reply apps/hostmetrics.go:314]: There should be no need, since it won't be enabled.

Comment (braydonk) [reply apps/hostmetrics.go:314]: Yeah it should be harmless to leave it here on both platforms, but might be confusing to future readers. The way this is written it's not elegant to exclude by just platform. What you'd need to do is change this to an array, add to the array conditionally based on platform, then spread it (with the `...` operator) into the arguments.

File: apps/hostmetrics.go [hunk 2]
```diff
@@ -31,7 +32,18 @@ func (r MetricsReceiverHostmetrics) Type() string {
 	return "hostmetrics"
 }
 
-func (r MetricsReceiverHostmetrics) Pipelines(_ context.Context) []otel.ReceiverPipeline {
+func (r MetricsReceiverHostmetrics) Pipelines(ctx context.Context) []otel.ReceiverPipeline {
+	p := platform.FromContext(ctx)
+	s := map[string]interface{}{
```
Comment (braydonk) [inline apps/hostmetrics.go:37]: `s` feels like a non-obvious name here. I'd recommend changing it.

**Review by braydonk — APPROVED** [review]

File: apps/hostmetrics.go (hunk 1 above)
Comment (igorpeshansky) [reply apps/hostmetrics.go:314]: It's confusing and pollutes the Linux configs, hence my suggestion.
As @braydonk said, you could do the following:
```
	transforms := []map[string]interface{}{
		// host transforms
		otel.RenameMetric(
			"system.cpu.time",
			"cpu/usage_time",
			…
		),
		…
		otel.RenameMetric(
			"system.paging.operations",
			"swap/io",
			…
		),
		// process transforms
		otel.RenameMetric(
			"process.cpu.time",
			"processes/cpu_time",
			…
		),
		…
		otel.RenameMetric(
			"process.memory.virtual",
			"processes/vm_usage",
			…
		),
	}
	if p.Type == platform.Windows {
		transforms = append(
			transforms, 
			otel.RenameMetric(
				"process.handles",
				"processes/windows/handles",
				otel.AddLabel("process", "all"),
			),
		)
	}
	transforms = append(
		transforms, 
		otel.AddPrefix("agent.googleapis.com"),
	)
	return []otel.ReceiverPipeline{{
		…
		Processors: map[string][]otel.Component{"metrics": {
			…
			otel.MetricsTransform(transforms...),
		}},
	}}
```

File: apps/hostmetrics.go [hunk 3]
```diff
@@ -76,6 +313,7 @@ func (r MetricsReceiverHostmetrics) Pipelines(ctx context.Context) []otel.Receiv
 					},
 				},
 			},
+			//transforms...,
```
Comment (igorpeshansky) [inline apps/hostmetrics.go:316]: Is this left over from earlier debugging?

Comment (avilevy18) [reply apps/hostmetrics.go]: Whoops, thanks for the catch.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Pull request merged**